From: Bhavana Nagendra <bnagendr@redhat.com>
Subject: RHEL5.1 [PATCH]: Add Greyhound performance counter events
Date: Fri, 23 Mar 2007 15:37:30 -0400
Bugzilla: 222126
Message-Id: <46042C7A.5090001@redhat.com>
Changelog: [x86]: Add Greyhound performance counter events


BZ 222126

AMD's new Greyhound processors have new performance counter events.  
This patch adds the necessary kernel-level support for these performance 
counter events to Oprofile.  There is a depending user
level oprofile patch that wcohen will incorporate into oprofile (BZ 
232956), once the oprofile-0.9.2-8 development version is available.

Basic functionality testing done, 'opcontrol -l' worked, then set up a 
counter and looked that what
it reported after running it. Also checked if NUMA is set up correctly 
by looking at the HT
bandwidth on each link (if the CPU is accessing local mem then there 
should be no traffic).

Please ACK, or comment.


--- linux-2.6.18.x86_64/arch/i386/oprofile/nmi_int.c.oporig	2007-03-20 13:29:09.000000000 -0400
+++ linux-2.6.18.x86_64/arch/i386/oprofile/nmi_int.c	2007-03-20 13:37:15.000000000 -0400
@@ -384,6 +384,10 @@ int __init op_nmi_init(struct oprofile_o
 				   user space an consistent name. */
 				cpu_type = "x86-64/hammer";
 				break;
+			case 0x10:
+				model = &op_athlon_spec;
+				cpu_type = "x86-64/family10";
+				break;
 			}
 			break;
  

